src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -444,7 +444,7 @@ def filter_devices(self, rook_pods: KubernetesResource, drive_group: DriveGroupS
                             all 
                             or (
                                 device.sys_api['node'] in matching_hosts
-                                and ((sizematcher != None) or sizematcher.compare(device))
+                                and ((sizematcher is None) or sizematcher.compare(device))
```

Comment:
i am far from an expert in rook mgr module, but i think the fix is correct.  and probably a better fix would be to initialize `sizematcher` with `AllMatcher('', None)` instead, like
```py
sizematcher: Matcher = AllMatcher('', None)
# ...
all 
or (device.sys_api['node'] in matching_hosts
    and sizematcher.compare(device)
    and ...
)
```

Reply:
Thanks, that suggestion is much clearer to read. I have pushed a change for that. I'll still wait for  a general thumbs-up/down from someone with rook module experience, at least for a day or two.

Reply:
@phlogistonjohn I think both your change and the improvement suggested by @tchaikov are correct. Anyway, as far as I know this code is not being used in rook latest versions anymore (it's part of OSDs creation which is now managed by the Rook orch).